Make shadow stack debuggable

jacob-hughes opened the pull request:
Debugging the shadow stack inside gdb is hard because it moves all the locals around breaking gdb assumptions. This patch makes it a little easier to debug by creating a 'debug local' for each stack frame which points to the frame's corresponding shadow stack offset. The 'shadowstack' pointer is then visible in gdb with 'info locals'.

This relies solely on LLVM's debug intrinsic library, so does not affect code generation when the '-g' flag is not set.

vext01 commented on llvm/lib/Transforms/Yk/ShadowStack.cpp:
```diff
@@ -147,6 +150,40 @@ class YkShadowStackImpl {
+    if (!SP) {
+      return;
+    }
+    DIBasicType *DISSTy = DIB->createBasicType("shadowstack", SFrameSize * 8,
```
What's the magic `8` here?

jacob-hughes replied:
`createBasicType` requires size in bits, `SFrameSize` is size in bytes.

vext01 commented:
When this has merged, can you please sync ykllvm in yk?

jacob-hughes commented:
Sorry, when I was preparing this I moved it into a separate function and forgot to guard the `DI->finalize()` behind a proper null check. I've fixed this here ([d44ac25](https://github.com/ykjit/ykllvm/pull/247/commits/d44ac25fea72f15af5a7a37266006e7fcbc3d4f3)). If you're happy it's ready to squash.

vext01 commented:
Go ahead.

jacob-hughes commented:
Squashed
